lib/content-services/src/lib/tree/services/tree.service.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { DataSource } from '@angular/cdk/collections';
+import { FlatTreeControl } from '@angular/cdk/tree';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { TreeNode } from '../models/tree-node.interface';
+import { TreeResponse } from '../models/tree-response.interface';
+
+@Injectable({ providedIn: 'root' })
+export abstract class TreeService<T extends TreeNode> extends DataSource<T>  {
+    public readonly treeControl: FlatTreeControl<T>;
+    public treeNodesSource = new BehaviorSubject<T[]>([]);
+
+    get treeNodes(): T[] {
+        // eslint-disable-next-line rxjs/no-subject-value
```

Comment:
https://stackoverflow.com/questions/37089977/how-to-get-current-value-of-rxjs-subject-or-observable

```
Important note from the author of RxJS 5: Using getValue() is a HUGE red flag you're doing something wrong. It's there as an escape hatch. Generally everything you do with RxJS should be declarative. getValue() is imperative. If you're using getValue(), there's a 99.9% chance you're doing something wrong or weird. – 
[Ben Lesh](https://stackoverflow.com/users/135786/ben-lesh)
 [Jul 21, 2017](https://stackoverflow.com/questions/37089977/how-to-get-current-value-of-rxjs-subject-or-observable#comment77420190_37089997)
```

Reply:
I got rid of it thanks!